apps/meteor/app/assets/server/assets.ts

```diff
@@ -260,29 +258,29 @@ class RocketChatAssetsClass {
 		await RocketChatAssetsInstance.deleteFile(asset);
 
 		const ws = RocketChatAssetsInstance.createWriteStream(asset, contentType);
-		ws.on('end', () => {
-			return setTimeout(async () => {
-				const key = `Assets_${asset}`;
-				const value = {
-					url: `assets/${asset}.${extension}`,
-					defaultUrl: assetInstance.defaultUrl,
-				};
-
-				void (async () => {
-					const { modifiedCount } = await Settings.updateValueById(key, value);
-					if (modifiedCount) {
-						void notifyOnSettingChangedById(key);
-					}
-				})();
-
-				return RocketChatAssets.processAsset(key, value);
-			}, 200);
-		});
 
-		rs.pipe(ws);
+		return new Promise((resolve) => {
+			ws.on('end', () => {
+				return setTimeout(async () => {
+					const key = `Assets_${asset}`;
+					const value = {
+						url: `assets/${asset}.${extension}`,
+						defaultUrl: assetInstance.defaultUrl,
+					} as IRocketChatAsset;
+
+					await RocketChatAssets.processAsset(key, value);
+					resolve({
+						key,
+						value,
+					});
+				}, 200);
+			});
+
+			rs.pipe(ws);
+		});
 	}
 
-	public async unsetAsset(asset: string): Promise<void> {
+	public async unsetAsset(asset: string) {
```

Comment:
Can we have the return typed here?

Reply:
I don't see much reason to type a return if it can be inferred automatically, unless I was working with an interface and wanted to enforce a contract(which is not the case here).